tools/c7n_azure/tests/test_sqldatabase.py

```diff
@@ -67,3 +70,208 @@ def _assert_found_only_test_database(self, resources):
         db = resources[0]
 
         self.assertEqual(db.get('name'), 'cctestdb')
+
+
+class BackupRetentionPolicyFilterTest(BaseTest):
+
+    def setUp(self):
+        super(BackupRetentionPolicyFilterTest, self).setUp()
+
+        self.resouce_group = "test_sqlserver"
+        self.parent_key = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/"
+        "test_sqlserver/providers/Microsoft.Sql/servers/cctestsqlserver"
+        self.name = "cctestdb"
+
+        manager = mock.Mock()
+        self.filter = BackupRetentionPolicyFilter('op-prop', 0, {}, manager)
+
+    def test_raises_key_error_without_resource_group(self):
+        with self.assertRaises(KeyError):
+            self.filter.process([{
+                ChildResourceQuery.parent_key: self.parent_key,
+                'name': self.name
+            }])
+
+    def test_raises_key_error_without_parent_key(self):
+        with self.assertRaises(KeyError):
+            self.filter.process([{
+                'resourceGroup': self.resouce_group,
+                'name': self.name
+            }])
+
+    def test_raises_key_error_without_name(self):
+        with self.assertRaises(KeyError):
+            self.filter.process([{
+                'resourceGroup': self.resouce_group,
+                ChildResourceQuery.parent_key: self.parent_key
+            }])
+
+    def test_raises_value_error_when_cannot_determine_sql_server_name(self):
+        with self.assertRaises(ValueError):
+            self.filter.process([{
+                'resourceGroup': self.resouce_group,
+                ChildResourceQuery.parent_key: "invalidResourceId",
+                'name': self.name
+            }])
+
+
+class ShortTermBackupRetentionPolicyFilterTest(BaseTest):
+
+    def test_validate_short_term_backup_retention_policy_filter_schema_validate(self):
+        with self.sign_out_patch():
+            p = self.load_policy({
+                'name': 'short-term-schema-validate',
+                'resource': 'azure.sqldatabase',
+                'filters': [
+                    {
+                        'type': 'short-term-backup-retention-policy',
+                        'op': 'gte',
+                        'retention-period-days': 60
+                    }
+                ]
+            }, validate=True)
+            self.assertTrue(p)
+
+    def test_find_database_with_short_term_retention_at_14_days(self):
+        p = self.load_policy({
+            'name': 'find-database-with-short-term-retention-at-14-days',
+            'resource': 'azure.sqldatabase',
```

Comment:
can we filter by name first? ( in other places too)
if we have multiple sql servers might not be reliable test

Reply:
with the exception of `test_filter_database_with_retention_period_unit_mismatch`, the list of resources received from the policy is being filtered by name already (after the policy is run, not within the policy definition).

But, I do think it would be a bit more readable to move that into the policy definition.